clients/src/main/java/org/apache/kafka/common/record/OffsetAndTimestamp.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.record;
```

Comment:
Thanks for the suggestion, we can put it in `org.apahce.kafka.clients.consumer`. My only concern is that `OffsetAndMetadata` is closely related to **committed offset** which is only used by consumers. So it makes sense to put it in the consumer package. On the other hand `OffsetAndTimestamp` seems associated with each individual message and therefore more common. But it looks on the producer side, we have separated the offset and timestamp without a container class. So maybe it is fine to put OffsetAndTimestamp into the consumer package.